voir/instruments/gpu/rocm.py

```diff
@@ -0,0 +1,192 @@
+        raise RuntimeError("Driver not initialized (amdgpu not found in modules)")
+
+
+def list_devices():
+    """Returns a list of GPU devices"""
+
+    numberOfDevices = c_uint32(0)
+    ret = rocmsmi.rsmi_num_monitor_devices(byref(numberOfDevices))
+
+    if rsmi_ret_ok(ret):
+        deviceList = list(range(numberOfDevices.value))
+        return deviceList
+    else:
+        exit(ret)
+
+
+def get_gpu_use(device):
+    """Return the current GPU usage as a percentage"""
+    percent = c_uint32()
+
+    ret = rocmsmi.rsmi_dev_busy_percent_get(device, byref(percent))
+
+    if rsmi_ret_ok(ret, device, "GPU Utilization "):
+        return percent.value
+
+    return -1
+
+
+def get_mem_info(device, memType="vram"):
+    """Return the specified memory usage for the specified device
+
+    @param device: DRM device identifier
+    @param type: [vram|vis_vram|gtt] Memory type to return
+    """
+    memType = memType.upper()
+    if memType not in memory_type_l:
+        raise RuntimeError("Invalid memory type %s" % (memType))
+
+    memoryUse = c_uint64()
+    memoryTot = c_uint64()
+    memUsed = None
+    memTotal = None
+
+    ret = rocmsmi.rsmi_dev_memory_usage_get(
+        device, memory_type_l.index(memType), byref(memoryUse)
+    )
+    if rsmi_ret_ok(ret, device, memType):
+        memUsed = memoryUse.value
+
+    ret = rocmsmi.rsmi_dev_memory_total_get(
+        device, memory_type_l.index(memType), byref(memoryTot)
+    )
+    if rsmi_ret_ok(ret, device, memType + " total"):
+        memTotal = memoryTot.value
+    return (memUsed, memTotal)
+
+
+def get_temp(device, sensor):
+    """Display the current temperature from a given device's sensor
+
+    @param device: DRM device identifier
+    @param sensor: Temperature sensor identifier
+    """
+    temp = c_int64(0)
+    metric = rsmi_temperature_metric_t.RSMI_TEMP_CURRENT
+    ret = rocmsmi.rsmi_dev_temp_metric_get(
+        c_uint32(device), temp_type_lst.index(sensor), metric, byref(temp)
+    )
+    if rsmi_ret_ok(ret, device, sensor, True):
+        return temp.value / 1000
+    return "N/A"
+
+
+def get_power(device):
+    """Return the current power level of a given device
+
+    @param device: DRM device identifier
+    """
+    power = c_uint32()
+    ret = rocmsmi.rsmi_dev_power_ave_get(device, 0, byref(power))
+    if rsmi_ret_ok(ret, device, "power"):
+        return power.value / 1000000
+    return "N/A"
+
+
+def is_available():
+    return IMPORT_ERROR is None
+
+
+def get_arch():
+    return "rocm"
+
+
+class Monitor:
+    def __init__(self) -> None:
+        if IMPORT_ERROR is not None:
+            raise IMPORT_ERROR
+
+        initialize_rsmi()
+        self.devices = list_devices()
+
+    def get_gpu_info(self, device):
+        util = get_gpu_use(device)
+        used, total = get_mem_info(device)
+        temp = get_temp(device, "edge")
+        power = get_power(device)
+
+        return {
+            "device": device,
+            "product": "ROCm Device",
```

Comment:
Can we do better than this? I couldn't find the GPU model with the other method, but maybe it's easier with the bindings.

Reply:
added the option